pkg/utils/applicationprofile.go

```diff
@@ -163,21 +163,21 @@ func GetApplicationProfileContainer(object *v1beta1.ApplicationProfile, containe
 	return nil
 }
 
-func MergePolicy(policy1, policy2 v1beta1.RulePolicy) v1beta1.RulePolicy {
+func MergePolicies(primary, secondary v1beta1.RulePolicy) v1beta1.RulePolicy {
 	mergedPolicy := v1beta1.RulePolicy{
-		AllowedContainer: policy1.AllowedContainer || policy2.AllowedContainer,
+		AllowedContainer: primary.AllowedContainer || secondary.AllowedContainer,
 	}
 
-	processMap := make(map[string]struct{})
+	processes := mapset.NewSet[string]()
 
-	for _, process := range policy1.AllowedProcesses {
-		processMap[process] = struct{}{}
+	for _, process := range primary.AllowedProcesses {
+		processes.Add(process)
 	}
-	for _, process := range policy2.AllowedProcesses {
-		processMap[process] = struct{}{}
+	for _, process := range secondary.AllowedProcesses {
+		processes.Add(process)
 	}
 
-	for process := range processMap {
+	for process := range processes.Iter() {
```

Comment:
I wonder if we want to sort AllowedProcesses... like we do with Syscalls https://github.com/kubescape/storage/blob/ebcd04e7b23cd48120e8a429eec8dc148a802cca/pkg/registry/file/applicationprofile_processor.go#L97

Reply:
this is to avoid sending too many new versions with the synchronizer if the content doesn't change, but just with a different order

Reply:
In storage or node-agent?

Reply:
here, the reason why we prune and sort the syscalls list in storage is that this list can be appended by different node-agents
actually maybe it's the case here too...

Reply:
Not sure I understand how the two cases are similar.
Rule Policies is container level and they suppose to occur rarely